File: Common/scriptbase.rb
require 'output'

#All scripts must inherit from this base file!

class ScriptBase
	
	attr_accessor :should_run
	attr_accessor :device_manager

	def r2_init

		@should_run = true
		@settings = @device_manager.get("settings")
		@log = @device_manager.get(@settings.i.logger)
		OP::set_logger @log
	end
	
	def enable
		@should_run = true
	end

	def disable
		@should_run = false
	end

end

File: Common/scriptbase.py
import clr
clr.AddReferenceToFileAndPath(r"R2Core.Scripting.dll")
import R2Core.Scripting
clr.ImportExtensions(R2Core.Scripting)

class ScriptBase:

	def d(self, message):
		self.logger.message(message)

	def e(self, message):
		self.logger.error(message)

	def w(self, message):
		self.logger.warning(message)

	def r2_init(self):
		self.logger = self.device_manager.Get("log")
